src/storage/core.rs

```diff
@@ -415,54 +417,97 @@ where
     /// Returns entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<Vec<Entry>> {
+    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<ReadResult<Vec<Entry>>> {
```

Comment:
I propose to add `read_all_with_deletion_marker` here too. `read_all` can call it and remove the last item if it is a deletion marker